src/megaapi_impl.cpp

```diff
@@ -23582,6 +23597,60 @@ void MegaApiImpl::queryGoogleAds(int adFlags, MegaHandle publicHandle, MegaReque
     waiter->notify();
 }
 
+void MegaApiImpl::setCookieSettings(int settings, MegaRequestListener *listener)
+{
+    MegaRequestPrivate* request = new MegaRequestPrivate(MegaRequest::TYPE_SET_ATTR_USER, listener);
+    request->setParamType(MegaApi::USER_ATTR_COOKIE_SETTINGS);
+    request->setNumDetails(settings);
+    requestQueue.push(request);
+    waiter->notify();
+}
+
+void MegaApiImpl::setCookieSettings_sendPendingRequests(MegaRequestPrivate* request)
+{
+    auto tmp = std::to_string(request->getNumDetails());
```

Comment:
Is this conversion really what we want to? I think it's not. We want to send the bytes corresponding to the integer passed by the app. We don't want a C-string representation of the number, do we?

Reply:
It may very well be so, however, it was an attempt to follow the code for `ATTR_RUBBISH_TIME` (a.k.a. _an integer 32 bits is the number of days for the auto-purge Rubbish bin ( ^!rubbishtime). It can be used as reference_), as indicated in the ticket description.